src/components/TextField/TextArea.test.js

```diff
@@ -0,0 +1,255 @@
+        textArea.simulate('mouseEnter');
+        expect(textArea.state('hasMouseOver')).equal(true);
+    });
+
+    it('sets state hasMouseOver to false when textarea sees mouseLeave', () => {
+        const textArea = createTextarea();
+        textArea.simulate('mouseEnter');
+        expect(textArea.state('hasMouseOver')).equal(true);
+        textArea.simulate('mouseLeave');
+        expect(textArea.state('hasMouseOver')).equal(false);
+    });
+
+    it('gives parent div focus class when textarea has focus ', () => {
+        const textArea = createTextarea();
+        textArea.find('textarea').simulate('focus');
+        expect(textArea).to.have.className('uir-textarea--focus');
+    });
+
+    it('renders an textarea element', () => {
+        const textArea = createTextarea();
+        expect(textArea.find('textarea').length).to.equal(1);
+    });
+
+    it('has the correct classes', () => {
+        const textArea = createTextarea();
+        expect(textArea).to.have.className('uir-textarea');
+        expect(textArea.find('label')).to.have.className('uir-textarea-label');
+        expect(textArea.find('textarea')).to.have.className('uir-textarea-input');
+    });
+
+    it('can pass in a className prop', () => {
+        const exampleClass = 'test-me';
+        const textArea = shallow(<TextArea className={exampleClass} />);
+        expect(textArea).to.have.className(exampleClass);
+    });
+
+    it('has a full width class if isFullWidth is passed', () => {
+        const textArea = shallow(<TextArea isFullWidth />);
+        expect(textArea).to.have.className('uir-textarea--full-width');
+    });
+
+    it('has a auto resize class if hasAutoHeight is passed', () => {
+        const textArea = shallow(<TextArea hasAutoHeight />);
+        expect(textArea).to.have.className('uir-textarea--has-auto-height');
+    });
+
+    it('shows a placeholder when textarea has focus', () => {
+        const examplePlaceholder = 'example placeholder';
+        const textArea = shallow(<TextArea placeholder={examplePlaceholder} />);
+        textArea.setState({ hasFocus: true });
+        const textarea = textArea.find('textarea');
+        expect(textarea.prop('placeholder')).to.equal(examplePlaceholder);
+    });
+
+    it('can take a starting value', () => {
+        const exampleValue = 'my example value';
+        const textArea = shallow(<TextArea value={exampleValue} />);
+        expect(textArea.find('textarea').prop('value')).to.equal(exampleValue);
+    });
+
+    it('updates the component state when changing the textarea value', () => {
+        const textArea = shallow(<TextArea />);
+        const instance = textArea.instance();
+        instance.inputRef = { value: 'test' };
+        const mock = sinon.mock(instance);
+        const expectation = mock.expects('setState');
+        textArea.find('textarea').simulate('change');
+        expect(expectation).to.be.called();
+        mock.restore();
```

Comment:
Btw, this should also work:

    const setStateSpy = sinon.spy(textArea.instance(), 'setState');
    ...
    expect(setStateSpy).to.have.been.called();

Reply:
Nice 👍